### PR TITLE
checking api header

### PR DESCRIPTION
**Pull request by 21ajayns**

### CODE REVIEW

**Review by jomolmp — APPROVED** [review]
All good. Approved!

File: app/Exceptions/ApiSecretException.php [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<?php
+
+namespace App\Exceptions;
+
+use Exception;
+
+class ApiSecretException extends Exception
```
Comment (boneymylady) [inline app/Exceptions/ApiSecretException.php]: Be more specific with naming - `MissingApiSecretException` or `ApiSecretNotFoundException`.

File: app/Http/Controllers/ApiHeaderController.php [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Response;
+use Illuminate\Http\Request;
+
+class ApiHeaderController extends Controller
+{
+    public function checkHealth(Request $request):Response 
+    {
+        $Value = $request->header();
```
Comment (boneymylady) [inline app/Http/Controllers/ApiHeaderController.php]: `$Value` or `$value`?

File: config/apisecret.php [hunk 3]
```diff
@@ -0,0 +1,5 @@
+<?php
+
+return [
+    'api-secret' => 'check-api-secret'
```
Comment (boneymylady) [inline config/apisecret.php]: Underscores (for keys) are usually followed as standard practice

File: app/Http/Middleware/ApiHeaderCheckMiddleware.php [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+            }
+            else
+            {
+                throw new ApiSecretException("Invalid key");
+            }
+        }
+        else
+        {
+            throw new ApiSecretException("Invalid header");
+        }
```
Comment (boneymylady) [inline app/Http/Middleware/ApiHeaderCheckMiddleware.php]: There is no problem in using `if` `else`. However, it is usually discouraged in the modern coding world due to hard to read. So you can short this code as 

```
if ($request->hasHeader('api-secret') && $request->header('api-secret') === $value) {
  return $next($request);
}
  throw new new ApiSecretException("Missing header credentials.");
```

File: tests/Feature/Http/Controllers/ApiHeaderControllerTest.php [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+        $response->assertStatus(200);
+
+        $expected = [
+            'api-secret' => [
+                'check-api-secret'
+            ]
+        ];
+
+        $response->assertJsonFragment($expected);
+    }
```
Comment (boneymylady) [inline tests/Feature/Http/Controllers/ApiHeaderControllerTest.php:25]: You have to write 3 more tests and make sure that exception is thrown (`$this->expectException()`):
1. `$value = ['api-secret' => null]`
2. `$value = ['api-secret' => '']`
3. `$value = []`

**Review by boneymylady — CHANGES_REQUESTED** [review]
Good attempt. A few changes.

File: tests/Feature/Http/Controllers/ApiHeaderControllerTest.php [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+
+    public function test_header_without_value()
+    {
+        $this->withoutExceptionHandling();
```
Comment (21ajayns) [inline tests/Feature/Http/Controllers/ApiHeaderControllerTest.php]: This line was commented out!! Forgot to add

File: tests/Feature/Http/Controllers/ApiHeaderControllerTest.php [hunk 7]
```diff
@@ -23,4 +23,43 @@ public function test_for_health_check()
...
+
+        $this->expectException(ApiSecretMissingException::class);
+        $this->expectExceptionMessage('Missing api header credentials');
+
+        $response = $this->json('GET', 'api/health-check',[], []);
+
+        $response->assertStatus(422);
+    }
```
Comment (boneymylady) [inline tests/Feature/Http/Controllers/ApiHeaderControllerTest.php]: As these three tests are doing the same assertion, you can provide the input via data provider. See the examples in repo.. More [examples](https://www.w3resource.com/php/PHPUnit/writing-tests-for-phpunit-data-providers.php) here.

**Review by boneymylady — CHANGES_REQUESTED** [review]

**Review by boneymylady — APPROVED** [review]

**Pull request merged**